src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -401,7 +401,7 @@ public GCFrameRegistration(void* allocation, uint elemCount, bool areByRefs = tr
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern unsafe void UnregisterForGCReporting(GCFrameRegistration* pRegistration);
 
-        internal static int GetHashCodeOfPtr(IntPtr ptr)
+        internal static int GetHashCodeOfPtr(nint ptr)
```

Comment:
This is one of the cases that is taking `IntPtr` but where `void*` is probably a "better" representation of what the code is doing.

It's not clear to me why we're doing all this logic over just doing `((nuint)(ptr)).GetHashCode()`

Reply:
Changing `IntPtr` parameter to `void*` has bigger impact, for example this method is called by passing different fields/properties with IntPtr type as argument, so the question is how to handle those? Are the also could/should be changed to `void*` or need to cast for each call?

Reply:
> It's not clear to me why we're doing all this logic over just doing ((nuint)(ptr)).GetHashCode()

This is hardening inherited from .NET Framework. The intention is to make hashcodes of reflection objects randomized to make them unpredictable.

This method is not used in many places after recent refactoring. I think it would be fine to delete it and replace the remaining instances with `Hash.Combine` that provides similar level of randomization. Hash.Combine is probably a bit slower than this method, but I think it is ok to take a small regression for the simplification.